sdk/core/azure-core/inc/azure/core.hpp

```diff
@@ -16,6 +16,7 @@
 #include "azure/core/datetime.hpp"
 #include "azure/core/dll_import_export.hpp"
 #include "azure/core/etag.hpp"
+#include "azure/core/match_conditions.hpp"
```

Comment:
Add `request_conditions.hpp` and unit test that verified simplified header correctness.